src/server/scene/basic_idle_hub.cpp

```diff
@@ -114,6 +117,17 @@ struct ms::BasicIdleHub::PendingRegistration
     std::set<IdleStateObserver const*> observers;
 };
 
+ms::BasicIdleHub::WakeLock::WakeLock(std::shared_ptr<BasicIdleHub> idle_hub)
+        : idle_hub{idle_hub}
+{
+}
+
+ms::BasicIdleHub::WakeLock::~WakeLock()
+{
+    idle_hub.lock()->alarm->reschedule_in(std::chrono::duration_cast<std::chrono::milliseconds>(
+            idle_hub.lock()->alarm_timeout.value()));
```

Comment:
You're calling `.lock()` on a `weak_ptr` and directly using that result. `.lock()` returns a `shared_ptr` to the object if it still exists, or a null `shared_ptr` if it doesn't. You can either use a `shared_ptr` to keep the idle hub alive as long as this object or you can:
```cpp
if (auto const shared_hub = idle_hub.lock())
{
    shared_hub->...
}
```
But you generally should not do `.lock()->` on a `weak_ptr`

Reply:
That *sure does* make more sense.